src/test/permission.spec.js

```diff
@@ -0,0 +1,129 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import jwt from 'jsonwebtoken';
+import app from '../index';
+
+chai.use(chaiHttp);
+
+describe('Test access based control functionality', () => {
+  it('should return 401 when user is not an admin or super admin', done => {
+    const newUser = {
+      id: 3,
+      username: 'Shola Bola',
+      email: 'sholabola@gmail.com',
+      role: 'user'
+    };
+    const userToken = jwt.sign(newUser, process.env.JWT_SECRET);
+    chai
+      .request(app)
+      .get('/api/v1/users')
+      .set('x-access-token', userToken)
+      .end((err, res) => {
+        expect(res.status).to.equal(401);
+        expect(res.body).to.have.property('status');
+        expect(res.body.error).to.equal(
+          'Unauthorized User, Please contact the administrator.'
+        );
+        done();
+      });
+  });
+
+  it('should return 200 when user is an admin or super admin', done => {
```

Comment:
The test descriptions don't tell what you're testing, making it impossible to tell their value add